src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs

```diff
@@ -24,8 +25,17 @@ internal abstract class AbstractClassificationService(ISyntaxClassificationServi
 {
     private readonly ISyntaxClassificationService _syntaxClassificationService = syntaxClassificationService;
 
-    private static Func<SyntaxNode, ImmutableArray<Classifiers.ISyntaxClassifier>>? s_getNodeClassifiers;
-    private static Func<SyntaxToken, ImmutableArray<Classifiers.ISyntaxClassifier>>? s_getTokenClassifiers;
+    private static ExtensionClassifierInfo? s_cachedExtensionClassifierInfo;
```

Comment:
alternatively.  each classification service is lang specific.  so don't cache these statically.  cache in each particular subclass.

Reply:
ooh, I like that better

Reply:
Bummer, RemoteSemanticClassificationService accesses this method statically.

Reply:
This seems like a common enough code path, that it should also hit the optimization, and I don't want to cache these both statically and on the instances, so I think I'd prefer to keep it as is unless you see a nice way around this.

Reply:
you can have RemoteSemanticClassificationService fetch the real service, not access it statically.

Reply:
I think the general idea works (I don't think RemoteSemanticClassificationService  can access AbstractClassificationService, but I can just have the static method do the cast check.